### ISSUE TITLE
Drupal 8 version?

### ISSUE THREAD

**Issue by cedewey** [MukurtuCMS/mukurtucms#108]
Hi there,

I have a potential client interested in using Mukurtu and I was curious if there are plans or interest in porting Mukurtu to Drupal 8. Also, I am part of the [Drutopia initiative](http://docs.drutopia.org) and I'd be interested in exploring the possibility of Mukurtu for Drupal 8 being built within the the Drutopia distro ecosystem.

**Comment by alexmerrill** [comment]
Hello --

There is certainly interest in porting Mukurtu to Drupal 8 but no immediate plans to do so.  We are trying to map out when the move to 8 makes the most sense in our overall development timeline and haven't been able to pin it down.  I had never heard of the Drutopia initiative before but, am very intrigued after a short survey of the info at the link you shared.  I would be interested in exploring the option as well.  From the site it sounds like Drutopia is in Alpha, is that still accurate?

**Comment by cedewey** [comment]
Hi @alexmerrill ,

That's right, it's in Alpha though a Beta is not far off. I think a Mukurtu distribution would be a wonderful addition to the Drutopia ecosystem. I'll email you off thread to discuss further. Thanks!

**Comment by kbighorse** [comment]
is there any update on Drupal 8 porting?

**Comment by steve-taylor-wsu** [comment]
It's in development currently with the goal of having some sort of early access for testing/feedback by the end of the year and a production release sometime in 2022.

**Comment by kbighorse** [comment]
Looks like Drupal 7 sunsets next year: https://www.drupal.org/about/drupal-7. It appears to be prudent to skip from Drupal 7 directly to Drupal 9: https://www.drupal.org/psa-2020-06-24. End-of-life for Drupal 8 is November of this year, well before that of Drupal 7 next year. Perhaps this issue should be closed in favor of porting to Drupal 9?

**Comment by kbighorse** [comment]
Looks like Drupal 10 won't be far behind:

> For security purposes, all Drupal 9 users will need to upgrade to Drupal 10 by November 2023.

**Comment by steve-taylor-wsu** [comment]
You are correct, internally we talk about "Drupal 8+" meaning anything above Drupal 7. By the time Mukurtu v4 reaches a production release, Drupal 8 will most likely have reached the end of it's support cycle.

With current development, Mukurtu v4 is being packaged as a Drupal install profile. Unless that changes, this means we will no longer be distributing Drupal ourselves, only the profile (we will have composer templates available to auto-install Drupal + Mukurtu). You'll be able to use it with any supported version of Drupal, including 9 and eventually 10.

**Comment by alexmerrill** [comment]
@kbighorse  Thanks for connecting, at the first link below we have some information from our newsletter about our plans for Drupal 8/9.  You might consider signing up if you haven't already (link to sign up is also below).
 
[Newsletter Announcment](https://mailchi.mp/d4641a11d465/mukurtu-3873713) 

[Mukurtu Newsletter signup](http://eepurl.com/bIwJqf)

**Comment by kbighorse** [comment]
Thank you! If I'm reading the Gantt chart correctly, Mukurtu CMS 3 supports Drupal 7 and Mukurtu CMS 4 will support Drupal 9? I'm not seeing anywhere it would make sense to support 8 and not 9.

I worked briefly with Drupal 7 and I just remember that the change to 8 was pretty big because D8 was built on Symfony going forward. And Symfony is driving the Drupal update schedule even now.

**Comment by steve-taylor-wsu** [comment]
That's correct, Mukurtu CMS v2 and v3 are basically the same. The version change is to signify a shift away from the old mobile app to the new mobile app. v4 is a complete rewrite and will support, as best we can on our timetable, the "current" version(s) of Drupal, whatever that may be at the time.

**Comment by andrewmallis** [comment]
The current version is still 3.x. 
The [December 2024](https://us12.campaign-archive.com/?u=0b0b343dc81346ec97cfc51d9&id=45075a8862) newsletter announces a mid-2025 release date.

The work seems to be happening here: https://github.com/MukurtuCMS/Mukurtu-CMS
